raft/rafttest/interaction_env_handler_campaign.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafttest
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/datadriven"
+)
+
+func (env *InteractionEnv) handleCampaign(t *testing.T, d datadriven.TestData) error {
+	idx := mustFirstAsInt(t, d) - 1
```

Comment:
This conversion from one-based indexing to zero-based seems error prone. Maybe it should be a map keyed by id instead? (would also allow for removal or non-sequential IDs)

Reply:
I took @nvanbenschoten's suggestion. Flexibility to allow bespoke IDs doesn't seem useful. If the occasional test wants a gap for some reason they can bootstrap some nodes and then never use them, which effectively does the same thing as those nodes not existing.